p2p/src/msg.rs

```diff
@@ -158,13 +154,28 @@ pub fn read_body<T: Readable>(h: &MsgHeader, stream: &mut dyn Read) -> Result<T,
 	ser::deserialize(&mut &body[..]).map_err(From::from)
 }
 
+/// Read (an unknown) message from the provided stream and discard it.
+pub fn read_discard(msg_len: u64, stream: &mut dyn Read) -> Result<(), Error> {
+	let mut buffer = vec![0u8; msg_len as usize];
```

Comment:
I wish we could skip allocation here, but I'm not sure we can do it with async io

Reply:
Yeah. We can always revisit this and optimize it later.
There might be something we can do down in read_exact so we're not allocating the whole thing like this.